sdk/core/azure-core-amqp/inc/azure/core/amqp/message_sender.hpp

```diff
@@ -13,131 +13,174 @@
+     *
+     */
+    std::string Name;
+
+    /** @brief The settle mode for the link associated with the message sender.
+     *
+     * This field indicates how the deliveries sent over the link SHOULD be settled. When this
+     * field is set to "mixed", the unsettled map MUST be sent even if it is empty. When this
+     * field is set to "settled", the value of the unsettled map MUST NOT be sent. See
+     * http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#doc-idp145616
+     * for more details.
+     *
+     */
+    SenderSettleMode SettleMode{};
+
+    /** @brief The source for the link associated with the message sender. */
+    Models::_internal::MessageSource MessageSource;
+
+    /** @brief The Maximum message size for the link associated with the message sender. */
+    Azure::Nullable<uint64_t> MaxMessageSize;
+
+    /** @brief The initial delivery count for the link associated with the message.
+     *
+     * The delivery-count is initialized by the sender when a link endpoint is created, and is
+     * incremented whenever a message is sent. Note that the value of this field can be
+     * overwritten by the remote peer if a link endpoint is attached with a transfer-id that
+     * indicates that the peer's value for the delivery-count is ahead of the sender's value.
+     *
+     */
+    Azure::Nullable<uint32_t> InitialDeliveryCount;
+
+    /** @brief If true, the message sender will log trace events. */
+    bool EnableTrace{false};
+  };
+
+  class MessageSender {
+  public:
+    using MessageSendCompleteCallback = std::function<void(
+        MessageSendStatus sendResult,
+        Azure::Core::Amqp::Models::AmqpValue const& deliveryState)>;
+
+    /** @brief Construct a message sender.
+     *
+     * @param session The AMQP session.
+     * @param target The target to which messages will be sent.
+     * @param options The options to use when sending messages.
+     * @param events The events handler for the message sender.
+     *
+     */
+    MessageSender(
+        Session const& session,
+        Models::_internal::MessageTarget const& target,
+        MessageSenderOptions const& options,
+        MessageSenderEvents* events);
+
+    /** @brief Construct a message sender for use in a message receiving handler.
+     *
+     * @param session The AMQP session.
+     * @param newLinkEndpoint The link endpoint to use for sending messages.
+     * @param target The target to which messages will be sent.
+     * @param options The options to use when sending messages.
+     * @param events The events handler for the message sender.
+     *
+     * @remarks This constructor is primarily used for testing scenarios.
+     */
+    MessageSender(
+        Session const& session,
+        LinkEndpoint& newLinkEndpoint,
+        Models::_internal::MessageTarget const& target,
+        MessageSenderOptions const& options,
+        MessageSenderEvents* events);
+
+    virtual ~MessageSender() noexcept;
```

Comment:
the class does not seem to have any other virtual functions - can it be made final and have virtual destructor removed?